content/post/azure-monitor-terraform.md

```diff
@@ -0,0 +1,230 @@
+---
+title: "Azure Monitor & Terraform: Collecting custom log files"
+date: 2023-10-03T14:46:23+01:00
+hero: /images/hero-3.jpg
+excerpt: "Working with Azure Monitor in collecting custom log files using Terraform"
+timeToRead: 5
+authors:
+  - Dylan McGinley
+draft: true
+license: cc-by
+---
+To collect custom log files from a VM (Virtual Machine) managed by Azure, it is recommended to use Azure Monitor and the Azure Monitor Agent (AMA). The Azure Monitor Agent is the successor to the Log Analytics Agent which is due to be deprecated on August 31, 2024. There are currently issues associated with the azurerm provider in relation to Azure Monitor.
+
+I suspect in the future these issues will be addressed with updates but for now this post will serve to provide a workaround along with an explanation as to what these are.
+
+Azurerm version 3.74.0
+
+The issue
+Currently the advised method of collecting logs and storing them with Azure Monitor relies on the following configuration
+
+* Setup a Log Analytics Workspace
+* Create a Log Analytics Workspace Table for storing logs
+* Install the correct agents on the VM
+* Setup a Data Collection Rule (DCR)
+* Setup a Data Collection Endpoint
+* Create a Data Collection Association between the Data Collection Rule and the VM
+
+The issue here pertains to the Log Analytics Workspace Table. According to the Azure documentation, it advises that a table be created in the Log Analytics Workspace to store the custom logs. Using the azurerm provider, It is implied that when creating the DCR and setting up the stream declaration that it will create the table in the Log Analytics Workspace, however this does not seem to be the case. Previously developers were able to create the table when setting up the Log Analytics Workspace
+
+```hcl
```

Comment:
do you think we should add more detail to the snapshot of the tf file? for example, where the file is located, the name of the files.

because the [following documentation](https://learn.microsoft.com/en-us/azure/templates/microsoft.operationalinsights/workspaces/tables?pivots=deployment-language-bicep#resource-format) you shared on this document pointed out where to go for the code/config snapshot.